### PR TITLE
Dev code refactor

### PR DESCRIPTION
**Pull request by luciferlinx101**
Refactored Code of the dev branch.

### CODE REVIEW

File: migrations/versions/2f98a4503e53_renamed_tokens_calls.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# revision identifiers, used by Alembic.
+revision = '2f98a4503e53'
+down_revision = '598cfb37292a'
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('agent_template_steps',
```
Comment (TransformerOptimus) [inline migrations/versions/2f98a4503e53_renamed_tokens_calls.py:21]: why are u recreating tables here?

File: migrations/versions/2f98a4503e53_renamed_tokens_calls.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    sa.Column('prompt', sa.Text(), nullable=True),
+    sa.Column('variables', sa.Text(), nullable=True),
+    sa.Column('output_type', sa.String(), nullable=True),
+    sa.Column('step_type', sa.String(), nullable=True),
+    sa.Column('next_step_id', sa.Integer(), nullable=True),
+    sa.Column('history_enabled', sa.Boolean(), nullable=True),
+    sa.Column('completion_prompt', sa.Text(), nullable=True),
+    sa.PrimaryKeyConstraint('id')
+    )
+    op.create_table('agent_templates',
```
Comment (TransformerOptimus) [inline migrations/versions/2f98a4503e53_renamed_tokens_calls.py:36]: this is not required.

File: migrations/versions/2f98a4503e53_renamed_tokens_calls.py (hunk 1 above)
Comment (luciferlinx101) [reply migrations/versions/2f98a4503e53_renamed_tokens_calls.py]: Fixed

File: migrations/versions/2f98a4503e53_renamed_tokens_calls.py (hunk 2 above)
Comment (luciferlinx101) [reply migrations/versions/2f98a4503e53_renamed_tokens_calls.py]: Removed

**Review by TransformerOptimus — APPROVED** [review]

**Pull request merged**